game/armedforces/forcegroup.py

```diff
@@ -0,0 +1,250 @@
+        units: set[UnitType[Any]] = set()
+        statics: set[Type[DcsUnitType]] = set()
+        for group in layout.groups:
+            for unit_type in group.possible_types_for_faction(faction):
+                if issubclass(unit_type, VehicleType):
+                    units.add(next(GroundUnitType.for_dcs_type(unit_type)))
+                elif issubclass(unit_type, ShipType):
+                    units.add(next(ShipUnitType.for_dcs_type(unit_type)))
+                elif issubclass(unit_type, StaticType):
+                    statics.add(unit_type)
+
+        return ForceGroup(
+            f"{layout.role.value}: {', '.join([t.description for t in layout.tasks])}",
+            list(units),
+            list(statics),
+            layout.role,
+            layout.tasks,
+            [layout],
+        )
+
+    def __str__(self) -> str:
+        return self.name
+
+    @classmethod
+    def named(cls, name: str) -> ForceGroup:
+        if not cls._loaded:
+            cls._load_all()
+        return cls._by_name[name]
+
+    def has_access_to_dcs_type(self, type: Type[DcsUnitType]) -> bool:
+        return (
+            any(unit.dcs_unit_type == type for unit in self.units)
+            or type in self.statics
+        )
+
+    def dcs_unit_types_for_group(self, group: GroupLayout) -> list[Type[DcsUnitType]]:
+        """TODO Description"""
+        unit_types = [t for t in group.unit_types if self.has_access_to_dcs_type(t)]
+
+        alternative_types = []
+        for accessible_unit in self.units:
+            if accessible_unit.unit_class in group.unit_classes:
+                unit_types.append(accessible_unit.dcs_unit_type)
+            if accessible_unit.unit_class in group.alternative_classes:
+                alternative_types.append(accessible_unit.dcs_unit_type)
+
+        return unit_types or alternative_types
+
+    def unit_types_for_group(self, group: GroupLayout) -> Iterator[UnitType[Any]]:
+        for dcs_type in self.dcs_unit_types_for_group(group):
+            if issubclass(dcs_type, VehicleType):
+                yield next(GroundUnitType.for_dcs_type(dcs_type))
+            elif issubclass(dcs_type, ShipType):
+                yield next(ShipUnitType.for_dcs_type(dcs_type))
+
+    def statics_for_group(self, group: GroupLayout) -> Iterator[Type[DcsUnitType]]:
+        for dcs_type in self.dcs_unit_types_for_group(group):
+            if issubclass(dcs_type, StaticType):
+                yield dcs_type
+
+    def random_dcs_unit_type_for_group(self, group: GroupLayout) -> Type[DcsUnitType]:
+        """TODO Description"""
+        return random.choice(self.dcs_unit_types_for_group(group))
+
+    def update_group(self, new_group: ForceGroup) -> None:
+        """Update the group from another group. This will merge statics and layouts."""
+        # Merge layouts and statics
+        self.statics = list(set(self.statics + new_group.statics))
+        self.layouts = list(set(self.layouts + new_group.layouts))
```

Comment:
It is not. Otherwise we would have many duplicate force_groups with just one layout. The idea here is to have a group of units which serve the same tasking with  a set of 1..N Layouts they can be generated. Otherwise we would have 1..N similar ForceGroups in regards to the actual units with exactly 1 Layout. I personally do not like it that way. It's not really clean tbh.